pi_top_usb_setup/extra/handle-usb-drive-for-pi-top-setup

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+###############################################################
+#                Unofficial 'Bash strict mode'                #
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/  #
+###############################################################
+set -euo pipefail
+IFS=$'\n\t'
+###############################################################
+
+log() {
+    if [ "$#" -gt 0 ]; then
+        msg="$1"
+    else
+        # Support reading from stdin
+        read msg
+    fi
+    logger -t pt-usb-setup "[$DEVICE | $OPERATION] $msg"
+}
+
+is_mounted() {
+    return $(mount | grep -qs "${DEVICE_NAME}")
+}
+
+mount_device() {
+    MOUNT_DST=${1:-}
+    if [ -z "${MOUNT_DST}" ]; then
+        log "No mount destination provided, exiting..."
+        exit 1
+    fi
+
+    if is_mounted "${DEVICE_NAME}"; then
+        log "Device ${DEVICE} is already mounted, skipping mount"
+        return
+    fi
+
+    log "Mounting ${DEVICE} into ${MOUNT_DST}"
+    mkdir -p "${MOUNT_DST}"
+    mount -t "${FS}" "${DEVICE}" "${MOUNT_DST}"
+    log "$(ls -l "${MOUNT_DST}")"
+}
+
+umount_device() {
+    MOUNT_DST=${1:-}
+    if [ -z "${MOUNT_DST}" ]; then
+        log "No umount destination provided, exiting..."
+        exit 1
+    fi
+
+    if is_mounted "${DEVICE_NAME}"; then
+        log "Unmounting ${MOUNT_DST}..."
+        sync
+        umount "${MOUNT_DST}"
+    else
+        log "Device ${DEVICE} is not mounted, skipping unmount"
+    fi
+
+    if [ -d "${MOUNT_DST}" ]; then
+        log "Removing ${MOUNT_DST}..."
+        rm -rf "${MOUNT_DST}"
+    fi
+}
+
+start_miniscreen_app() {
+    log "Starting miniscreen application..."
+    systemctl start pt-usb-setup
+}
+
+has_pitop_offline_setup_files() {
+    if [ -f "${MOUNT_POINT}/pitop-usb-setup.tar.gz" ]; then
+        return 0
+    fi
+    return 1
+}
+
+remount_to_final_location() {
+    # Remount to location where our offline apt source will look for updates
+    umount_device "${MOUNT_POINT}"
+    mount_device "${FINAL_MOUNT_POINT}"
+}
+
+has_valid_format() {
+    # TODO: check if device has a valid format
+    return 0
+}
+
+remove_systemd_drop_in() {
+  rm -rf /lib/systemd/system/pt-usb-setup.service.d/*.conf
+}
+
+handle_add() {
+    mount_device "${MOUNT_POINT}"
+
+    if has_pitop_offline_setup_files; then
+        remount_to_final_location
+        remove_systemd_drop_in
+        start_miniscreen_app
```

Comment:
maybe instead of using a systemd service for the restart this bash script could block until the python exits and then have an exit code based restart thing:
```suggestion
        start_miniscreen_app
        needs_restart=$? == 10
        if needs_restart: start_miniscreen_app
```

Reply:
not sure if this is better or worse, but it does remove the systemd component of this so maybe simpler? up to you? would it be more or less reliable? the bash process is hanging around over the python either way (before the restart). this way there is always a bash script controlling the python rather than switching over to systemd

Reply:
one issue with this is if udev has a timeout that kills this process - we think it does